### DISCUSSION TITLE
Defer static page generation until first page access

### DISCUSSION THREAD

**Discussion by sspenst** [vercel/next.js#39052] [Ideas]
### Describe the feature you'd like to request

We are building our next.js application using docker, so we don't have database access during build time. This results in next.js generating a static page with our mock data instead of the actual data, which means our users will see incorrect data until the next revalidation occurs on this page.

### Describe the solution you'd like

Similar to dynamic routes with `fallback: true`, if there was an option to avoid generating the static page during the build and instead generate the page on the first page access, it would solve this issue.

### Describe alternatives you've considered

getServerSideProps would solve this issue, but it adds a lot of unnecessary server bandwidth for a page that updates infrequently.

**Comment by ijjk** [comment]
Hi, if you'd like to leverage `fallback: true` to skip generation during build you can migrate the non-dynamic pages to dynamic pages instead which could then leverage `fallback: true`.

**Reply by sspenst** [reply]
Thanks, I've actually done that for now. I changed the endpoint from `/catalog` to `/catalog/[index]`, and I return a blank page unless `index === 'all'`. Feels like a bit of a roundabout way to do it but gets the job done in the meantime.

**Comment by sspenst** [comment]
Thought I'd update this thread because I found a cleaner way to implement `fallback: true`.

Create your page in a directory like `/pages/<PAGENAME>/[[...route]]/index.tsx`, then use the following structure:

```
export async function getStaticPaths() {
  return {
    paths: [],
    fallback: true,
  };
}

export async function getStaticProps(context: GetServerSidePropsContext) {
  if (context.params?.route) {
    return { notFound: true };
  }

  return {
    props: {
      ...
    },
    revalidate: 60,
  };
}
```

This avoids having to check for a specific route like I was doing before (eg checking `index === 'all'`), instead you just filter out every route that isn't the base route.
